whisper/whisperv5/whisper.go

```diff
@@ -177,12 +181,23 @@ func (w *Whisper) GetIdentity(pubKey string) *ecdsa.PrivateKey {
 }
 
 func (w *Whisper) GenerateSymKey(name string) error {
-	buf := make([]byte, aesKeyLength*2)
-	_, err := crand.Read(buf) // todo: check how safe is this function
+	const size = aesKeyLength * 2
+	buf := make([]byte, size)
+	buf2 := make([]byte, size)
+	_, err := crand.Read(buf)
 	if err != nil {
 		return err
 	} else if !validateSymmetricKey(buf) {
-		return fmt.Errorf("crypto/rand failed to generate random data")
+		return fmt.Errorf("error in GenerateSymKey: crypto/rand failed to generate random data")
+	}
+
+	randomize(buf2)
+	if !validateSymmetricKey(buf2) {
+		return fmt.Errorf("error in GenerateSymKey: math/rand failed to generate random data")
+	}
+
+	for i := 0; i < size; i++ {
```

Comment:
From crypto/rand documentation https://golang.org/pkg/crypto/rand/:
> Package rand implements a cryptographically secure pseudorandom number generator.
> On Linux, Reader uses getrandom(2) if available, /dev/urandom otherwise. On OpenBSD, Reader uses getentropy(2). On other Unix-like systems, Reader reads from /dev/urandom. On Windows systems, Reader uses the CryptGenRandom API.

Urandom is cryptographically secure, and does not block. It's the Right Choice (tm). http://www.2uo.de/myths-about-urandom/ 

I agree with @Arachnid's points.